### PR TITLE
Remove custom IServiceScopeProvider in favor of using default scope accessible via ControllerContext.HttpContext.RequestServices

### PR DESCRIPTION
**Pull request by korDen**
There is no need to create a new scope since Controller is already created inside one.
This also simplifies the code because request handlers no longer need to capture IServiceScopeProvider in their ctor.

### CODE REVIEW

File: TRANSMUTANSTEIN/PreAuthHandlerTest.cs [hunk 1]
```diff
@@ -1,35 +1,61 @@
...
+
+        IFeatureCollection features = new FeatureCollection();
+        features.Set<IServiceProvidersFeature>(serviceProvidersFeature);
+
+        DefaultHttpContext httpContext = new DefaultHttpContext(features);
+        ControllerContext controllerContext = new ControllerContext();
+        controllerContext.HttpContext = httpContext;
+
+        return controllerContext;
+    }
```
Comment (mrhappyasthma) [inline TRANSMUTANSTEIN/PreAuthHandlerTest.cs:34]: Should we pull this to a test helper utils style file? I imagine we'll need to construct these for basically every test.

Comment (korDen) [reply TRANSMUTANSTEIN/PreAuthHandlerTest.cs]: Yes, this was just for a proof of concept. Extracted into a ControllerContextForTesting class that can be instantiated (and eventually customized).

File: ZORGATH/PreAuthHandler.cs [hunk 2]
```diff
@@ -1,4 +1,6 @@
-﻿namespace ZORGATH;
+﻿using Microsoft.Extensions.DependencyInjection;
```
Comment (mrhappyasthma) [inline ZORGATH/PreAuthHandler.cs:1]: Can we drop this or move it to the global Usings.cs?

**Review by mrhappyasthma — APPROVED** [review]
LGTM after the final comment

File: ZORGATH/PreAuthHandler.cs (hunk 2 above)
Comment (korDen) [reply ZORGATH/PreAuthHandler.cs]: Done

**Pull request merged**